oak/server/rust/oak_runtime/src/node/grpc/server.rs

```diff
@@ -60,38 +51,31 @@ impl GrpcServerNode {
     /// Creates a new [`GrpcServerNode`] instance, but does not start it.
     ///
     /// `channel_writer` is initialized with `None`, and is filled in at `run`-time.
-    pub fn new(
-        node_name: &str,
-        runtime: RuntimeProxy,
-        address: SocketAddr,
-        tls_identity: Identity,
-        initial_reader: oak_abi::Handle,
-    ) -> Self {
+    pub fn new(node_name: &str, address: SocketAddr, tls_identity: Identity) -> Self {
         Self {
             node_name: node_name.to_string(),
-            runtime,
             address,
             tls_identity,
-            initial_reader,
             channel_writer: None,
         }
     }
 
-    /// Reads an [`oak_abi::Handle`] from a channel specified by [`GrpcServerNode::initial_reader`].
+    /// Reads an [`oak_abi::Handle`] from a channel specified by `handle`.
     /// Returns an error if couldn't read from the channel or if received a wrong number of handles
     /// (not equal to 1).
-    fn init_channel_writer(&mut self) -> Result<(), OakStatus> {
-        let read_status = self
-            .runtime
-            .wait_on_channels(&[self.initial_reader])
-            .map_err(|error| {
-                error!("Couldn't wait on the initial reader handle: {:?}", error);
-                OakStatus::ErrInternal
-            })?;
+    fn init_channel_writer(
+        &mut self,
+        runtime: &RuntimeProxy,
```

Comment:
Not sure I fully agree with the changes in commit 1d20c1bc62b708c296c35c87bde125d13f307b66 . As much as I like functional idioms, this seems to introduce an artificial distinction between the runtime proxy and all the other fields. I mean, you can take this to the extreme and have an empty struct and pass everything as function args, but it kind of defeats the point of encapsulation.

Reply:
As well as merging four different copies of the same thing (pass proxy+handle at construction, store in per-Node-type struct field, make use of those field (only) in `run()`), I also think of it as tightening up the scope of the things available to the Node: if the Node is not currently running, then it should not need access to a `Runtime` instance, nor be able to do any messaging using a Handle.